cmd/zc_traverser_blob_account.go

```diff
@@ -61,27 +63,40 @@ func (t *blobAccountTraverser) listContainers() ([]string, error) {
 				return nil, err
 			}
 			for _, v := range resp.ContainerItems {
-				// Match a pattern for the container name and the container name only.
-				if t.containerPattern != "" {
-					if ok, err := containerNameMatchesPattern(*v.Name, t.containerPattern); err != nil {
-						// Break if the pattern is invalid
-						return nil, err
-					} else if !ok {
-						// Ignore the container if it doesn't match the pattern.
-						continue
+				// a nil list also returns 0
+				if len(t.cachedContainers) == 0 {
+					// Match a pattern for the container name and the container name only.
+					if t.containerPattern != "" {
+						if ok, err := containerNameMatchesPattern(*v.Name, t.containerPattern); err != nil {
+							// Break if the pattern is invalid
+							return nil, err
+						} else if !ok {
+							// Ignore the container if it doesn't match the pattern.
+							continue
+						}
 					}
 				}
 
-				cList = append(cList, *v.Name)
+				// get a list of containers that are not excluded
+				if len(t.excludeContainerName) > 0 {
+					so := StoredObject{ContainerName: *v.Name}
+					for _, f := range t.excludeContainerName {
+						if !f.DoesPass(so) {
+							// Ignore the container if the container name should be excluded
+							continue
```

Comment:
Should we log this as info that we are excluding this container. User has given the names in the list but the log will help us identify what all we excluded due to filter match.

Reply:
This is a great idea! I added it so the run looks like this:

```
INFO: Scanning...
INFO: Any empty folders will not be processed, because source and/or destination doesn't have full folder support
INFO: Skipped container(s): new, stuff, test, test-stuff, test2
```

Reply:
See added changes in line 112.